### ISSUE TITLE
feat: improve segment design in ios

### ISSUE THREAD

**Issue by xnousnow** [ionic-team/ionic-framework#26825]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

- [ ] The label in the segment should get smaller when you press and hold it.
- [ ] The white switch that indicates the selection (I don't know what can I call that) when you press and hold it is different:
![image](https://user-images.githubusercontent.com/119923091/220047998-a04cfe6a-b477-4d36-9059-18777627f12d.png)
Although the feature is good, I found a difference between native and ionic.

### Describe the Use Case

This is just a design suggestion, so the use case is the same.

### Describe Preferred Solution

Resolving the above issue.

**Comment by liamdebeasi** [comment]
Thanks for the report. Can you please clarify the issue you are seeing? The segment button in Ionic scales down when you press and hold.

**Comment by DwieDima** [comment]
yeah i can confirm the scale on press is slightly larger on `ion-segment`.

IOS normal
![IMG_4755](https://user-images.githubusercontent.com/26873275/220144781-bfee829b-7834-4a81-93f6-76cd4eff75b9.jpg)


IOS pressed
![IMG_4758](https://user-images.githubusercontent.com/26873275/220144830-2733542c-f3f6-402f-9c42-fc5e2a0f22f0.jpg)


Ionic normal
![IMG_4756](https://user-images.githubusercontent.com/26873275/220144798-502487b8-2cbf-4b59-aa2e-fc1a80499f68.jpg)


Ionic pressed
![IMG_4757](https://user-images.githubusercontent.com/26873275/220144814-a8dd37ce-0777-4011-a2f1-ee5abb2d6bea.jpg)

**Comment by xnousnow** [comment]
For clarify:

https://user-images.githubusercontent.com/119923091/220217342-3a4f5918-7a1a-479e-a658-f3f12c006144.MP4

**Comment by DwieDima** [comment]
> yeah i can confirm the scale on press is slightly larger on `ion-segment`. Also `ion-segment-button` borders are not aligned with `ion-segment` border when pressed like on ios. Furthermore the text inside `ion-segment-button` doesnt transforms on change.
> 
> IOS normal ![IMG_4755](https://user-images.githubusercontent.com/26873275/220144781-bfee829b-7834-4a81-93f6-76cd4eff75b9.jpg)
> 
> IOS pressed ![IMG_4758](https://user-images.githubusercontent.com/26873275/220144830-2733542c-f3f6-402f-9c42-fc5e2a0f22f0.jpg)
> 
> Ionic normal ![IMG_4756](https://user-images.githubusercontent.com/26873275/220144798-502487b8-2cbf-4b59-aa2e-fc1a80499f68.jpg)
> 
> Ionic pressed ![IMG_4757](https://user-images.githubusercontent.com/26873275/220144814-a8dd37ce-0777-4011-a2f1-ee5abb2d6bea.jpg)

Hi @liamdebeasi, i have edited my comment to also include segment button text transform on change to my description. Will this also be included in this ticket?

https://user-images.githubusercontent.com/26873275/220342194-b6b5ec4a-4c3f-4de4-9982-53f98b265b5a.mov

**Comment by liamdebeasi** [comment]
We will align with the iOS spec as closely as possible when working on this task.
